### ISSUE TITLE
[BUG]: Model Training "checkpoint_frequency:" parameter ineffective in YAML file (Torch)

### ISSUE THREAD

**Issue by roshanr11** [CosmiQ/solaris#353]
## Summary of the bug
The `checkpoint_frequency` parameter for training the model is ineffective in the YAML intended to be supplied to the Solaris CLI/Python API. 

## Steps to reproduce the bug

```
checkpoint_frequency: 5
```

1. Complete every step of the Solaris training/testing process necessary, until you get to the creation of the YAML file
2. When creating your YAML file, specify the `checkpoint_frequency` parameter under `training` to a number of your choice -- let's say 5. 
3. Simply run the deep-learning pipeline process on your setup, perhaps with the Solaris CLI command `solaris_run_ml -c /path/to/your_file.yml`

## Buggy behavior and/or error message
Observe that the models are checkpointed by any epochs with  *lesser validation loss* than previously seen. See screenshots for more info.

## Expected behavior
The model should have checkpointed every 5 epochs, consistently, instead of solely by the epochs with the least validation loss. 

## Screenshots

<img width="408" alt="Screen Shot 2020-06-22 at 8 53 01 AM" src="https://user-images.githubusercontent.com/55853065/85289662-c8b19800-b465-11ea-880f-332609b900b6.png">

In the instance of this setup, the following specification was used in the YAML file `checkpoint_frequency: 1`. We observe that the model paths stored are not checkpointed every epoch, but, in fact, are checkpointed by successively lower validation loss each time.

**Comment by roshanr11** [comment]
Pull request merged - #355
